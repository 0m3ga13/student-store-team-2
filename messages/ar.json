{
    "Index": {
        "Title": "المتجر الصغير",
        "Home": "الرئيسية",
        "Products": "المنتجات",
        "Orders": "الطلبيات",
        "State": "الحالة",
        "Buy": "شراء",
        "Borrow": "إعارة",
        "Description": "هذه المنصة هي مربحة للجميع: القرضاء يحققون دخلاً إضافيًا، والمقترضون يقللون التكاليف.",
        "Profile": "الحسـاب",
        "For Sale": "للبيع",
        "For Borrow": "للإعارة",
        "Sale & Borrow": "إعارة وشراء",
        "Day": "لليوم",
        "How It Works": "طريـقة العمل",
        "Hello": "مرحبا",
        "Sign out": "تسجيل الخروج",
        "Sign In": "تسجيل الدخول",
        "Login with Google": "سحل الدخول بحسابك",
        "Sign Up With Google": "سحل الدخول بحسابك"
    }
}
