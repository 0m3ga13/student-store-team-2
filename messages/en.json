{
    "Index": {
        "Title": "MiniStore",
        "Home": "Home",
        "Products": "Products",
        "Orders": "Orders",
        "State": "State",
        "Donate": "Donate",
        "Buy": "Buy",
        "Borrow": "Borrow",
        "Description": "This platform is a win-win for everyone involved lenders earn extra income, borrowers cut costs.",
        "Profile": "Profile",
        "for_sale": "for_sale",
        "for_borrow": "for_borrow",
        "Sale & Borrow": "Sale & Borrow",
        "Day": "Day",
        "How It Works": "How It Works",
        "recently added": "recently added",
        "Hello": "Hello",
        "Sign out": "Sign out",
        "Sign In": "Sign In",
        "Login with Google": "Login with Google",
        "Sign Up With Google": "Sign Up With Google",
        "Free delivery": "Free delivery",
        "feature descriprion 1": "You can receive your order without incurring any additional shipping costs",
        "qualite guarantee": "qualite guarantee",
        "feature descriprion 2": "Our products or services meet high standards and perform as promised.",
        "daily offers": "daily offers",
        "feature descriprion 3": "Exclusive discounts and promotions on a daily basis",
        "100% secure payement": "100% secure payement",
        "feature descriprion 4": "Your financial information is protected and your transactions are safe from unauthorized access",
        "My Dashboard": "My Dashboard",
        "Awaiting Pickup": "Awaiting Pickup",
        "All": "All",
        "Active": "Active",
        "Completed": "Completed",
        "Sort": "Sort:",
        "Newest": "Newest",
        "Oldest": "Oldest",
        "my stuff":"my stuff",
        "borrowings": "borrowings",
        "lendings": "lendings",
        "purchases": "purchases",
        "sales": "sales",
        "My Borrowings": "My Borrowings",
        "My Lendings": "My Lendings",
        "My Purchases": "My Purchases",
        "My Sales": "My Sales",
        "General details": "General details",
        "First Name": "First Name",
        "Last Name": "Last Name",
        "Birth Date": "Birth Date",
        "Gender": "Gender",
        "Institution": "Institution",
        "Contact & address Detail": "Contact & address Detail",
        "Phone Number": "Phone Number",
        "Building": "Building",
        "Street": "Street",
        "City": "City",
        "State_place": "State",
        "Save Changes": "Save Changes",
        "Upload Image": "Upload Image",
        "Close": "Close",
        "Upload": "Upload",
        "Loading": "Loading",
        "Not a member ?": "Not a member ?",
        "Sign Up": "Sign Up",
        "Store": "Store",
        "Create Your": "Create Your",
        "and start exchanging with other students": "and start exchanging with other students",
        "Or Login Using:": "Or Login Using:",
        "Login": "Login",
        "LOGIN": "LOGIN ",
        "Already a member ?": "Already a member ?",
        "Email and password are required": "Email and password are required",
        "Password should be at least 8 characters long": "Password should be at least 8 characters long",
        "and Conditions and Privacy Policy of The website": "and Conditions and Privacy Policy of The website",
        "Email": "Email",
        "Password": "Password",
        "Full Name": "Full Name",
        "Address": "Address",
        "and Privacy Policy": "and Privacy Policy",
        "I have read and agree with all of the Terms": "I have read and agree with all of the Terms",
        "PROFILE": "PROFILE",
        "SECURITY": "SECURITY",
        "Upload ID": "Upload ID",
        "Edit ID": "Edit ID",
        "Edit Image": "Edit Image",
        "WHY DONATE": "WHY DONATE",
        "MAKE A DIFFERENCE BY DONATING": "MAKE A DIFFERENCE BY DONATING",
        "Donate paragraph": "In our fast-paced lives, it's easy to forget the power of gratitude. Students, often on tight budgets, rely on our support. Your donation is more than just financial aid; it's a symbol of gratitude. By giving generously, you're helping students create a brighter future, one contribution at a time.",
        "Small Help": "Small Help",
        "Some Help": "Some Help",
        "Big Help": "Big Help",
        "List an Item": "List an Item",
        "Drag and drop some files here or click to select files": "Drag and drop some files here or click to select files",
        "Location": "Location",
        "Category": "Category",
        "books": "books",
        "kitchen": "kitchen",
        "sports": "sports",
        "games": "games",
        "clothes": "clothes",
        "electronics": "electronics",
        "furniture": "furniture",
        "description": "Description",
        "Product Condition": "Product Condition",
        "Sell": "Sell",
        "Price Per Week": "Price Per Week",
        "Price": "Price",
        "Upload Image(s)": "Upload Image(s)",
        "List": "List",
        "Product Name": "Product Name",
        "Comments": "Comments",
        "Type your comment": "Type your comment",
        "Type your reply": "Type your reply",
        "Respond": "Respond",
        "Like": "Like",
        "Send": "Send",
        "Authentication error": "Invalid login credentials. Please check your email and password.",
        "Filter": "Filter",
        "Pending": "Pending",
        "Quick Links": "Quick Links",
        "About LACITÉ": "ABOUT LACITÉ",
        "About Us": "ABOUT US",
        "Github Project Link": "GITHUB PROJECT LINK",
        "Contact Us": "Contact Us",
        "We Couldn't Find the page you're looking for": "We Couldn't Find the page you're looking for",
        "Go To Home": "Go To Home",
        "We Couldn't Find the requested Product": "We Couldn't Find the requested Product."
    }
}
