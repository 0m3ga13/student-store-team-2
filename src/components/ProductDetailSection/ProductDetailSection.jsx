"use client";
import React from "react";
import ProfilePic from "../ProfilePicture/ProfilePic";
import ProfileUserInfo from "../profileUserInformation/ProfileUserInfo";
import BuyBorrowBtn from "../buyborrow-btn/BuyBorrowBtn";
import { useTranslations } from "next-intl";
import Image from "next/image";
import Link from "next/link";
import ContactDetails from "../contactDetails/ContactDetails/ContactDetails";

export default function ProductDetailSection({ user, productData }) {
    const t = useTranslations("Index");
    return (
        <section className=' ml-2 w-full flex flex-col'>
            <div className='w-1/2 m-4 flex gap-2'>
                <ProfilePic user={user} />
                <ProfileUserInfo user={user} />
            </div>
            <div className='border border-0.5 border[#EDF1F3] rounded-md drop-shadow-sm p-4 w-10/12'>
                <div className='mb-10 ml-4'>
                    <h2 className='font-lato text-3xl uppercase text-[#55585B]'>
                        {productData
                            ? productData.name
                            : "Product Name Not Available"}
                    </h2>
                    <div className='relative'>
                        <Image
                            className='z-0'
                            src='/conditionLabelVector.svg'
                            alt='Item Condition'
                            width={139}
                            height={39}
                        ></Image>
                        <div className='font-jost font-extrabold text-base text-bkg absolute bottom-1 left-2 tracking-wider '>
                            {t("State")}:{" "}
                            {productData
                                ? productData.condition + "/10"
                                : "Condition Not Available"}
                        </div>
                    </div>
                </div>

                <p className='w-5/6 text-xs font-medium mb-2 leading-5 tracking-wide'>
                    {productData
                        ? productData.description
                        : "Description Not Available"}
                </p>
                <div className='ml-8 mt-8 text-[#3A3A3A] w-full flex items-center gap-10 my-1'>
                    <div className='w-[180px] flex flex-row '>
                        <BuyBorrowBtn label='Buy' />
                        <p className='font-lato text-lg font-semibold'>
                            $
                            {productData
                                ? productData.price.sell_price
                                : " Sell Price Not Available"}
                        </p>
                    </div>
                    <div className='w-[188px] flex flex-row'>
                        <BuyBorrowBtn label='Borrow' />
                        <p className='font-lato text-lg font-semibold'>
                            $
                            {productData
                                ? productData.price.borrow_price
                                : "Borrow Price Not Available"}
                            <span className='text-[#55585B] text-sm font-thin font-jost'>
                                per day
                            </span>
                        </p>
                    </div>
                </div>
                <div className=' w-full flex mt-8 gap-2'>
                    <p className=' text-xs text-[#3A3A3A] font-semibold '>
                        Category :
                    </p>
                    <Link
                        href='/'
                        className='font-normal text-[#72AEC8] underline hover:text-[#45adda] text-xs transform hover:scale-105 transition-transform duration-300 ease-in-out'
                    >
                        {productData
                            ? productData.category
                            : "Category Not Available"}{" "}
                    </Link>
                </div>
                <ContactDetails user={user} />
            </div>
        </section>
    );
}
