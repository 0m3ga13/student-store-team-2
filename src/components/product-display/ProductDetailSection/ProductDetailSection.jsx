"use client";
import React from "react";

// import ItemConditionLabel from "../../productCard/itemConditionLabel/ItemConditionLabel";
import BuyBorrowBtn from "../../buyborrow-btn/BuyBorrowBtn";

// Placeholder object to be replaced with data from Firestore
const productDetails = {
    name: "circle dining table",
    description:
        "Tristique ullamcorper nunc egestas non. Justo, cum feugiat imperdiet nulla molestie ac vulputate scelerisque amet. Bibendum adipiscing platea blandit sit sed quam semper rhoncus. Diam ultrices maecenas consequat eu tortor. Orci, cras lectus mauris, cras egestas quam venenatis neque.",
    condition: "6",
    price: {
        borrow_price: "20",
        sell_price: "100",
    },
    categoty: "Fourniture",
};

export default function ProductDetailSection() {
    return (
        <section className=' w-full flex flex-col'>
            <div className='mb-10 ml-4'>
                <h2 className='font-lato text-5xl uppercase text-[#55585B]'>
                    {productDetails.name}
                </h2>
                {/* <div className='absolute bottom-0 left-0 mt-6 text-black'>
            <ItemConditionLabel
              itemCondition={productDetails.condition + "/10"}
              />
        </div> */}

                <span classeName='bg-[#FFBD00] text-white font-jost font-semibold  '>
                    note: {productDetails.condition + "/10"}
                </span>
            </div>

            <p className='w-5/6 text-base font-medium mb-2'>
                {productDetails.description}
            </p>
            <div className='ml-8 text-[#3A3A3A] w-full flex items-center gap-10 my-1'>
                <div className='w-2/12 flex flex-row '>
                    <BuyBorrowBtn label='Buy' />
                    <p className='font-lato text-lg font-semibold'>
                        ${productDetails.price.sell_price}
                    </p>
                </div>
                <div className='w-[188px] flex flex-row'>
                    <BuyBorrowBtn label='Borrow' />
                    <p className='font-lato text-lg font-semibold'>
                        ${productDetails.price.borrow_price}{" "}
                        <span className='text-[#55585B] text-sm font-thin font-jost'>
                            per day
                        </span>
                    </p>
                </div>
            </div>
        </section>
    );
}
