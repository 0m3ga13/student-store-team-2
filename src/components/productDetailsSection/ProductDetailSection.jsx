"use client";
import Image from "next/image";
import Link from "next/link";
import { useRouter } from "next/navigation";
import { useTranslations } from "next-intl";
import React, { useEffect } from "react";
import { useState } from "react";
import { FaSpinner } from "react-icons/fa";
import { toast } from "react-toastify";

import {
    createBorrow,
    createPurchase,
    hasBorrowed,
    hasPurchased,
    setNotification,
} from "@/lib/_supabase";

import BuyBorrowBtn from "../buyborrow-btn/BuyBorrowBtn";
import ContactDetails from "../contactDetails/ContactDetails/ContactDetails";
import ProfilePic from "../profilePicture/ProfilePic";
import ProfileUserInfo from "../profileUserInformation/ProfileUserInfo";
import { useUser } from "../userProvider/UserProvider";

const sendItemRequestNotification = async (
    requester,
    owner,
    action,
    itemType
) => {
    const statusMessage = {
        buy: "has requested to buy your item",
        borrow: "has requested to borrow your item",
    };

    const data = {
        message: statusMessage[action],
        path: itemType === "buy" ? "sales" : "lendings", // Adjust path based on item type
    };

    const res = await setNotification(requester, owner, data, "order");
    return res;
};

export default function ProductDetailSection({ user, product }) {
    const t = useTranslations("Index");
    const router = useRouter();
    const currentUser = useUser();

    const [isLoading, setLoading] = useState(false);
    const [hasUserPurchased, setHasUserPurchased] = useState(false);
    const [hasUserBorrowed, setHasUserBorrowed] = useState(false);
    const [isChecking, setIsChecking] = useState(false);
    useEffect(() => {
        const checkUserPurchase = async () => {
            setIsChecking(true);
            if (currentUser.user && product) {
                const userHasPurchased = await hasPurchased(
                    product.pid,
                    currentUser.user.id
                );
                await setHasUserPurchased(userHasPurchased);
                await setHasUserBorrowed(userHasPurchased);
                const userHasBorrowed = await hasBorrowed(
                    product.pid,
                    currentUser.user.id
                );
                setHasUserBorrowed(userHasBorrowed);
            }
            setIsChecking(false);
        };

        checkUserPurchase();
    }, [isLoading]);

    const handleClick = async () => {
        try {
            if (currentUser.user) {
                setLoading(true);
                if (product?.offer_type === "for_sale") {
                    const order = {
                        productId: product.pid,
                        receiver: currentUser.user.id,
                        sender: product.uid,
                    };

                    await sendItemRequestNotification(
                        order.receiver,
                        order.sender,
                        "buy",
                        "buy"
                    );

                    await createPurchase(order);
                } else {
                    const order = {
                        productId: product.pid,
                        receiver: currentUser.user.id,
                        sender: product.uid,
                    };

                    await sendItemRequestNotification(
                        order.receiver,
                        order.sender,
                        "borrow",
                        "borrow"
                    );
                    await createBorrow(order);
                }
                toast("Item Added Successfully", {
                    hideProgressBar: true,
                    autoClose: 2000,
                    type: "success",
                    position: "bottom-center",
                });
            } else {
                toast("You must be logged In to Continue", {
                    hideProgressBar: false,
                    autoClose: 2000,
                    type: "error",
                    position: "bottom-center",
                });
                router.push("/sign-in");
            }
        } catch (error) {
            toast("You've already Sent a Request", {
                hideProgressBar: true,
                autoClose: 2000,
                type: "error",
                position: "bottom-center",
            });
        } finally {
            setLoading(false);
        }
    };
    return (
        <section className='w-full sm:w-full flex flex-col'>
            <div className='w-full m-4 flex gap-2'>
                <ProfilePic user={user} />
                <ProfileUserInfo user={user} />
            </div>
            <div className='border border-0.5 border[#EDF1F3] rounded-md drop-shadow-sm p-4 w-full'>
                <div className='mb-10 ml-4'>
                    <h2 className='font-lato text-lg sm:3xl uppercase text-[#55585B]'>
                        {product && product.name}
                    </h2>
                    <div className='relative mt-2'>
                        <Image
                            className='z-0'
                            src='/conditionLabelVector.svg'
                            alt='Item Condition'
                            width={139}
                            height={39}
                        ></Image>
                        <div className='font-jost font-extrabold text-sm sm:text-base text-bkg absolute bottom-1 left-2 tracking-wider '>
                            {t("State")}: {product && product.condition + "/10"}
                        </div>
                    </div>
                </div>

                <p className='w-full sm:w-5/6 text-sm sm:text-base font-medium mb-2 leading-5 tracking-wide'>
                    {product && product.description}
                </p>
                <div className=' mt-8 text-[#3A3A3A] w-full flex flex-col items-start sm:flex-row sm:items-center my-1'>
                    {isChecking ? (
                        <FaSpinner className='h-8 w-8 animate-spin duration-150 text-accent' />
                    ) : (
                        <>
                            {product?.offer_type === "for_sale" ? (
                                <div className=' flex flex-row gap-2 w-full sm:w-2/3 '>
                                    <BuyBorrowBtn
                                        label='Buy'
                                        handleClick={handleClick}
                                        disabled={hasUserPurchased}
                                        isLoading={isLoading}
                                    />
                                    {hasUserPurchased ? (
                                        <div className='font-lato text-md font-semibold'>
                                            Check your{" "}
                                            <Link
                                                className='underline text-accent'
                                                href='/dashboard'
                                            >
                                                Dashboard
                                            </Link>
                                        </div>
                                    ) : (
                                        <p className='font-lato text-lg font-semibold'>
                                            ${product?.sale_offer?.price}
                                        </p>
                                    )}
                                </div>
                            ) : (
                                <div className='flex flex-row gap-2 sm:gap-0 w-full sm:w-2/3 mt-2'>
                                    <BuyBorrowBtn
                                        label='Borrow'
                                        handleClick={handleClick}
                                        disabled={hasUserBorrowed}
                                        isLoading={isLoading}
                                    />
                                    {hasUserBorrowed ? (
                                        <div className='font-lato text-md font-semibold'>
                                            Check your{" "}
                                            <Link
                                                className='underline text-accent'
                                                href='/dashboard'
                                            >
                                                Dashboard
                                            </Link>
                                        </div>
                                    ) : (
                                        <p className='font-lato text-lg font-semibold '>
                                            ${product?.borrow_offer?.price}
                                            <span className='text-[#55585B] text-sm font-thin font-jost pl-2'>
                                                per day
                                            </span>
                                        </p>
                                    )}
                                </div>
                            )}
                        </>
                    )}
                </div>
                <div className=' w-full flex mt-8 gap-2'>
                    <p className=' text-sm sm:text-base text-[#3A3A3A] font-semibold '>
                        {t("Category:")}
                    </p>
                    <Link
                        href={`/products?category=${product?.categories?.category_name}`}
                        className='font-normal text-[#72AEC8] underline hover:text-[#45adda] text-sm sm:text-base transform hover:scale-105 transition-transform duration-300 ease-in-out'
                    >
                        {product?.categories?.category_name}
                    </Link>
                </div>
                <ContactDetails user={user} />
            </div>
        </section>
    );
}
