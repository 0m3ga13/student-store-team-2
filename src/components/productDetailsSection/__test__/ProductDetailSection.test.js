import { NextIntlClientProvider } from "next-intl";
import renderer from "react-test-renderer";

import ProductDetailSection from "../ProductDetailSection";
import messages from "../../../../messages/en.json";
const user = { phone_num: "0556214761" };
jest.mock("next/navigation", () => ({
    useRouter: jest.fn(() => ({
        push: jest.fn(),
    })),
    useSearchParams: jest.fn(() => ({
        get: jest.fn(),
    })),
    usePathname: jest.fn(),
    useParams: () => ({
        locale: "en",
    }),
}));
it("renders correctly", () => {
    const tree = renderer
        .create(
            <NextIntlClientProvider locale='en' messages={messages}>
                <ProductDetailSection user={user} />
            </NextIntlClientProvider>
        )
        .toJSON();
    expect(tree).toMatchSnapshot();
});
