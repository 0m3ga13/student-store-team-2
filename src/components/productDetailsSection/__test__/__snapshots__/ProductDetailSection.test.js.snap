// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<section
  className="w-full sm:w-full flex flex-col"
>
  <div
    className="w-full m-4 flex gap-2"
  >
    <div
      className="w-16 h-16 bg-sky-200 rounded-full flex items-center justify-center drop-shadow"
      style={
        Object {
          "background": "lavender",
          "backgroundSize": "cover",
        }
      }
    />
    <div
      clssName="flex flex-col p-2 ml-4 border border-red-300 border-l-1 "
    >
      <h2
        className="font-semibold text-base text-[#55585B]"
      >
         
      </h2>
      <div
        className="text-[#55585B] flex gap-1"
      >
        <svg
          fill="currentColor"
          height="1em"
          stroke="currentColor"
          strokeWidth="0"
          style={
            Object {
              "color": undefined,
            }
          }
          viewBox="0 0 512 512"
          width="1em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M496 128v16a8 8 0 0 1-8 8h-24v12c0 6.627-5.373 12-12 12H60c-6.627 0-12-5.373-12-12v-12H24a8 8 0 0 1-8-8v-16a8 8 0 0 1 4.941-7.392l232-88a7.996 7.996 0 0 1 6.118 0l232 88A8 8 0 0 1 496 128zm-24 304H40c-13.255 0-24 10.745-24 24v16a8 8 0 0 0 8 8h464a8 8 0 0 0 8-8v-16c0-13.255-10.745-24-24-24zM96 192v192H60c-6.627 0-12 5.373-12 12v20h416v-20c0-6.627-5.373-12-12-12h-36V192h-64v192h-64V192h-64v192h-64V192H96z"
          />
        </svg>
        <p
          className="font-medium text-xs  pt-1"
        />
      </div>
      <div
        className="text-[#72ADC7] flex gap-1"
      >
        <svg
          fill="currentColor"
          height="1em"
          stroke="currentColor"
          strokeWidth="0"
          style={
            Object {
              "color": undefined,
            }
          }
          viewBox="0 0 256 256"
          width="1em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M128,24a80,80,0,0,0-80,80c0,72,80,128,80,128s80-56,80-128A80,80,0,0,0,128,24Zm0,112a32,32,0,1,1,32-32A32,32,0,0,1,128,136Z"
            opacity="0.2"
          />
          <path
            d="M200,224H150.54A266.56,266.56,0,0,0,174,200.25c27.45-31.57,42-64.85,42-96.25a88,88,0,0,0-176,0c0,31.4,14.51,64.68,42,96.25A266.56,266.56,0,0,0,105.46,224H56a8,8,0,0,0,0,16H200a8,8,0,0,0,0-16ZM56,104a72,72,0,0,1,144,0c0,57.23-55.47,105-72,118C111.47,209,56,161.23,56,104Zm112,0a40,40,0,1,0-40,40A40,40,0,0,0,168,104Zm-64,0a24,24,0,1,1,24,24A24,24,0,0,1,104,104Z"
          />
        </svg>
        <p
          className="font-normal text-xs "
        />
      </div>
    </div>
  </div>
  <div
    className="border border-0.5 border[#EDF1F3] rounded-md drop-shadow-sm p-4 w-full"
  >
    <div
      className="mb-10 ml-4"
    >
      <h2
        className="font-lato text-lg sm:3xl uppercase text-[#55585B]"
      />
      <div
        className="relative mt-2"
      >
        <img
          alt="Item Condition"
          className="z-0"
          data-nimg="1"
          decoding="async"
          height={39}
          loading="lazy"
          onError={[Function]}
          onLoad={[Function]}
          src="/conditionLabelVector.svg"
          style={
            Object {
              "color": "transparent",
            }
          }
          width={139}
        />
        <div
          className="font-jost font-extrabold text-sm sm:text-base text-bkg absolute bottom-1 left-2 tracking-wider "
        >
          State
          :
           
        </div>
      </div>
    </div>
    <p
      className="w-full sm:w-5/6 text-sm sm:text-base font-medium mb-2 leading-5 tracking-wide"
    />
    <div
      className=" mt-8 text-[#3A3A3A] w-full flex flex-col items-start sm:flex-row sm:items-center my-1"
    >
      <div
        className=" flex flex-row gap-2 w-full sm:w-2/3 "
      >
        <div
          className=" sm:w-1/3 w-1/2 flex justify-evenly flex-col"
        >
          <button
            className="bg-[#0989FF] transition ease-in-out delay-150 hover:-translate-y-1 hover:scale-110 hover:bg-[#026fce] duration-300 text-white text-center font-lato text-base px-7 py-1 rounded shadow-md"
          >
            Buy
          </button>
        </div>
        <p
          className="font-lato text-lg font-semibold"
        >
          $
        </p>
      </div>
      <div
        className="flex flex-row gap-2 sm:gap-0 w-full sm:w-2/3 mt-2"
      >
        <div
          className=" sm:w-1/3 w-1/2 flex justify-evenly flex-col"
        >
          <button
            className="bg-[#FFBD00] transition ease-in-out delay-150 hover:-translate-y-1 hover:scale-110 hover:bg-[#e8ac09] duration-300 text-white text-center hover:transition-all font-lato text-base px-5 py-1 rounded shadow-md sm-absolute"
          >
            Borrow
          </button>
        </div>
        <p
          className="font-lato text-lg font-semibold "
        >
          $
          <span
            className="text-[#55585B] text-sm font-thin font-jost pl-2"
          >
            per day
          </span>
        </p>
      </div>
    </div>
    <div
      className=" w-full flex mt-8 gap-2"
    >
      <p
        className=" text-sm sm:text-base text-[#3A3A3A] font-semibold "
      >
        Category :
      </p>
      <a
        className="font-normal text-[#72AEC8] underline hover:text-[#45adda] text-sm sm:text-base transform hover:scale-105 transition-transform duration-300 ease-in-out"
        href="/"
        onClick={[Function]}
        onMouseEnter={[Function]}
        onTouchStart={[Function]}
      />
    </div>
    <div
      className="mt-8"
    >
      <h1
        className="font-lato text-md font-semibold text-[#55585B]"
      >
        Contact Details
      </h1>
      <div
        className="ml-4 mt-3 font-jost flex gap-1 text-[#72ADC7]"
      >
        <svg
          fill="currentColor"
          height="1em"
          stroke="currentColor"
          strokeWidth="0"
          style={
            Object {
              "color": undefined,
            }
          }
          viewBox="0 0 384 512"
          width="1em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M97.333 506.966c-129.874-129.874-129.681-340.252 0-469.933 5.698-5.698 14.527-6.632 21.263-2.422l64.817 40.513a17.187 17.187 0 0 1 6.849 20.958l-32.408 81.021a17.188 17.188 0 0 1-17.669 10.719l-55.81-5.58c-21.051 58.261-20.612 122.471 0 179.515l55.811-5.581a17.188 17.188 0 0 1 17.669 10.719l32.408 81.022a17.188 17.188 0 0 1-6.849 20.958l-64.817 40.513a17.19 17.19 0 0 1-21.264-2.422zM247.126 95.473c11.832 20.047 11.832 45.008 0 65.055-3.95 6.693-13.108 7.959-18.718 2.581l-5.975-5.726c-3.911-3.748-4.793-9.622-2.261-14.41a32.063 32.063 0 0 0 0-29.945c-2.533-4.788-1.65-10.662 2.261-14.41l5.975-5.726c5.61-5.378 14.768-4.112 18.718 2.581zm91.787-91.187c60.14 71.604 60.092 175.882 0 247.428-4.474 5.327-12.53 5.746-17.552.933l-5.798-5.557c-4.56-4.371-4.977-11.529-.93-16.379 49.687-59.538 49.646-145.933 0-205.422-4.047-4.85-3.631-12.008.93-16.379l5.798-5.557c5.022-4.813 13.078-4.394 17.552.933zm-45.972 44.941c36.05 46.322 36.108 111.149 0 157.546-4.39 5.641-12.697 6.251-17.856 1.304l-5.818-5.579c-4.4-4.219-4.998-11.095-1.285-15.931 26.536-34.564 26.534-82.572 0-117.134-3.713-4.836-3.115-11.711 1.285-15.931l5.818-5.579c5.159-4.947 13.466-4.337 17.856 1.304z"
          />
        </svg>
        <p
          className="text-sm font-medium"
        >
           
          Phone Number Not Available
        </p>
      </div>
      <div
        className="ml-4 mt-1 font-jost flex gap-1 text-[#72ADC7]"
      >
        <svg
          fill="currentColor"
          height="1em"
          stroke="currentColor"
          strokeWidth="0"
          style={
            Object {
              "color": undefined,
            }
          }
          viewBox="0 0 24 24"
          width="1em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M0 0h24v24H0z"
            fill="none"
          />
          <path
            d="M20 4H4c-1.1 0-1.99.9-1.99 2L2 18c0 1.1.9 2 2 2h16c1.1 0 2-.9 2-2V6c0-1.1-.9-2-2-2zm0 4l-8 5-8-5V6l8 5 8-5v2z"
          />
        </svg>
        <p
          className="text-sm font-medium"
        >
           
          Email Not Available
        </p>
      </div>
    </div>
  </div>
</section>
`;
